### PR TITLE
Add CAD_to_OpenMC surface mesh creation package.

### PR DESCRIPTION
**Pull request by ebknudsen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ebknudsen** [comment]
@conda-forge/help-python ready for review I believe

File: recipes/cad-to-openmc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+{% set version = "0.3.3a2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  sha256: 15689aff401b196a8b2a95b7571dab9dc328c4b70fd79f38fa812e806fe5c986
+  url: https://github.com/openmsr/CAD_to_OpenMC/archive/refs/tags/v{{ version }}.tar.gz
+build:
```
Comment (ocefpaf) [inline recipes/cad-to-openmc/meta.yaml:11]:
```suggestion

build:
```

File: recipes/cad-to-openmc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/cad-to-openmc/meta.yaml:12]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/cad-to-openmc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    - {{ compiler('c') }}
+  host:
+    - python
+    - setuptools >=61.0
+    - pip
+  run:
+    - python >=3.9
+    - gmsh
+    - python-gmsh
+    - moab >=5.3.0
```
Comment (ocefpaf) [inline recipes/cad-to-openmc/meta.yaml:27]: You can skip Windows b/c this package is not available there.
BTW, do you want to check that macOS error?

Comment (ebknudsen) [reply recipes/cad-to-openmc/meta.yaml:28]: I do want to check the MacOS error but not know as such. I think it is a bigger upstream issue though, so I think I'll have to punt a present. I.e. I should perhaps skip it. I tried only building on linux (f280da2) but that failed. should that have been [linux64] perhaps?

Comment (ocefpaf) [reply recipes/cad-to-openmc/meta.yaml:28]: Can you add skips for Windows and macOS then?

File: recipes/cad-to-openmc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  version: {{ version }}
+
+source:
+  sha256: 15689aff401b196a8b2a95b7571dab9dc328c4b70fd79f38fa812e806fe5c986
+  url: https://github.com/openmsr/CAD_to_OpenMC/archive/refs/tags/v{{ version }}.tar.gz
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  skip: True  # [not linux64]
```
Comment (ocefpaf) [inline recipes/cad-to-openmc/meta.yaml:15]:
```suggestion
  skip: True  # [not linux]
```

File: recipes/cad-to-openmc/meta.yaml (hunk 3 above)
Comment (ebknudsen) [reply recipes/cad-to-openmc/meta.yaml:29]: Sure - can do. I have tried two things: skipping by means of "not linux" and "not linux_64". The former fails, whereas the latter works. AFAICT the fail is on python 3.12rc which is not done for the latter.
I'll try skipping win and osx explicitly, but I don't think that's the issue here. Not quite sure what the sane way of doing this is now - I shall have to run a few more tests.

Comment (ebknudsen) [reply recipes/cad-to-openmc/meta.yaml:31]: Have added skips for win and osx as well as skips for python < 3.8 & >3.12. osx have to come later as I don't have access to infrastructure for that right now. Python version problems are upstream problems which I can't do much about I'm afraid

**Pull request merged**